manifest-db/centos_9-x86_64-edge_commit_rt-boot.json

```diff
@@ -8096,5 +8096,6 @@
         }
       }
     }
-  }
-}
\ No newline at end of file
+  },
+  "unwanted_tmpfiled_d":["rpm-ostree-1-autovar.conf"]
```

Comment:
Nitpick: I find this name a bit misleading. I would suggest something like `skip_tmpfilesd_paths`. Also the name of the property contains a typo.

In addition to that, I wonder how do you plan to handle updating manifests from composer, since I expect this to get dropped on manifest update. I think that it would be better to provide this as an argument to the test in `.gitlab-ci.yml` for the specific distro runner instead of storing it in the manifest.

If there are good reasons to keep it in the manifest, I would propose to nest this under a new top-level object such as `"image_test_config"` or something similar. We may end up adding more and it will be nicer to have them all nested under one umbrella object.

Reply:
I good for the naming change :)
updating won't erase the value, only the manifest part and the image-info get changed. It's isn't stored at manifest level but a level higher in the metadata parts of the DB. I can create a section metadata to hold this kind of stuff if you'd prefer.

Reply:
My preference is still to not have it defined in the image test case (what I meant by the manifest in my message, I mean basically the whole file), but provided to the test case as an argument when comparing image-info reports.

My point is that the current way it most probably works is:

1. Generate a new image test with image-info for a new image type or a new distro.
2. Then you need to manually change the image test and add any configuration for the test case.
3. Only now the image test will work.

So there is always a potential manual step for each new image types or new distros. While if it was passed as an argument to all c9s image tests, then you would not need to do anything for new image types or distros.

Lastly, using 'metadata` for the image test case configuration is IMO misleading, since it is really a configuration for the test case, not some additional data about it.

Reply:
I'm not a huge fan of having to set this as a parameter on the command line tbh. The reason why is that it scatters the information all around. Having it in the database makes it clearer and easier to find again. If we were to set the list on the command line in the .gitlabci yaml file, we would have to set it twice. One for MDB's ci and  one for osbuild's ci, with the implication of harder maintenance.

We already have metadata in the DB entry, like the image type or the arch. For me it makes sense to embed the `skip_tmpfilesd_paths` as it is a specificity of the entry itself. The DB being there to hold the necessary information to test image production, so containing in itself the data to filter at comparison time feels straightforward to me.

```
{
  "id": "centos_9-aarch64-ami-boot",
  "desc": {
    "distro": "centos-9",
    "arch": "aarch64",
    "image-type": "ami"
  },
  "manifest": {
  },
  "image-info": {
  }
}
```

There is still the question of the addition of new distributions that will not have this parameter set to them. And I think it's fine to add it manually when necessary. This PR also adds full testing of the DB on CI push, which means that after an update is proposed, it's gonna be tested as osbuild would be. Meaning that, we will detect early the need of adding tmpfilesd to skip to the list.

Reply:
> I'm not a huge fan of having to set this as a parameter on the command line tbh. The reason why is that it scatters the information all around. Having it in the database makes it clearer and easier to find again. If we were to set the list on the command line in the .gitlabci yaml file, we would have to set it twice. One for MDB's ci and one for osbuild's ci, with the implication of harder maintenance.

OK, that's a fair point.

> We already have metadata in the DB entry, like the image type or the arch. For me it makes sense to embed the `skip_tmpfilesd_paths` as it is a specificity of the entry itself. The DB being there to hold the necessary information to test image production, so containing in itself the data to filter at comparison time feels straightforward to me.
> 
> ```
> {
>   "id": "centos_9-aarch64-ami-boot",
>   "desc": {
>     "distro": "centos-9",
>     "arch": "aarch64",
>     "image-type": "ami"
>   },
>   "manifest": {
>   },
>   "image-info": {
>   }
> }
> ```

Wow, I was not aware of the fact that the image test manifests stored in `manifest-db` reinvented their actual content, compared to what we store in osbuild-composer. E.g. I was not aware that the `compose-request` https://github.com/osbuild/osbuild-composer/blob/ed4d12fe33e2eb2d09e4cd70e3e550ff8861d75d/test/data/manifests/centos_8-aarch64-ami-boot.json#L2-L22 is completely dropped in the db entry and that some values from it are instead specified under a completely different object. Not sure what was the decision of loosing the information about the compose request that was used to generate the manifest (without it there is no easy way to generate the same manifest again from composer). This is all sufficient as is for osbuild testing, but can potentially limit the usefulness of the db entry in the future. Not an issue related to this PR, but I wanted to express my surprise.

I do not feel strongly about embedding image test configuration (I still don't think that these are metadata) in the db entry itself. I think that this is OK and makes things easier when the same  configuration needs to be used in osbuild CI as well as this repo's CI.

> There is still the question of the addition of new distributions that will not have this parameter set to them. And I think it's fine to add it manually when necessary. This PR also adds full testing of the DB on CI push, which means that after an update is proposed, it's gonna be tested as osbuild would be. Meaning that, we will detect early the need of adding tmpfilesd to skip to the list.

This all sounds good. You could also define this only in the manifest-db `.gitlab-ci.yaml`, which would result in the value being embedded in the db entry (which would be then used as is in osbuild). But that's more of an enhancement that can be added later. I'm not a big fan of manual steps in workflows that can be automated, but if you are willing to do it, I'm fine with it.

Reply:
> Wow, I was not aware of the fact that the image test manifests stored in manifest-db reinvented their actual content, compared to what we store in osbuild-composer. E.g. I was not aware that the compose-request https://github.com/osbuild/osbuild-composer/blob/ed4d12fe33e2eb2d09e4cd70e3e550ff8861d75d/test/data/manifests/centos_8-aarch64-ami-boot.json#L2-L22 is completely dropped in the db entry and that some values from it are instead specified under a completely different object.

:smile: We would have to ask Christian what he had in mind for this decision ^^

Yeah so compose request is dropped, and we only keep the few bits necessary to run the testing. One major advantage of this is that the DB can be tailored to what we want it to be. And it's kind of easy to add new elements in it if we need to. I'm not that worried about it's upgradability.

The update workflow assumes that it exists a function to generate the manifests elsewhere. Now this function is the one from composer, e.g, the first step of the update process is to fun the gen_manifests program, once it's done, the manifests are embedded in the DB and a build is triggered. Once the build is finished, image-info data is generated and included in the DB, then a PR is opened.

> This all sounds good. You could also define this only in the manifest-db .gitlab-ci.yaml, which would result in the value being embedded in the db entry (which would be then used as is in osbuild). But that's more of an enhancement that can be added later. I'm not a big fan of manual steps in workflows that can be automated, but if you are willing to do it, I'm fine with it.

I'm not a huge fan of doing a bot's job either, I'm sure we will find a proper way to update the fields automatically as the need happen.